tests/unit/handlers/guildRoleHandler.test.ts

```diff
@@ -0,0 +1,264 @@
+    expect(response.status).toBe(404);
+    expect(jsonResponse).toEqual(responseConstants.NOT_FOUND);
+  });
+
+  it("should return not found object if there is no roles with given name in guild", async () => {
+    getGuildRoleByNameSpy.mockResolvedValueOnce(undefined);
+
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      params: {
+        roleName: "everyone",
+      },
+      headers: { Authorization: "Bearer testtoken" },
+    });
+
+    const response: JSONResponse = await getGuildRoleByRoleNameHandler(
+      mockRequest,
+      guildEnv
+    );
+    const jsonResponse: { error: string } = await response.json();
+    expect(response.status).toBe(404);
+    expect(jsonResponse).toEqual(responseConstants.NOT_FOUND);
+  });
+
+  it("should return role fetch failed error if there was an error while fetching roles", async () => {
+    getGuildRoleByNameSpy.mockRejectedValueOnce({
+      message: responseConstants.ROLE_FETCH_FAILED,
+    });
+
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      params: {
+        roleName: "everyone",
+      },
+      headers: { Authorization: "Bearer testtoken" },
+    });
+
+    const response: JSONResponse = await getGuildRoleByRoleNameHandler(
+      mockRequest,
+      guildEnv
+    );
+    const jsonResponse: { roles: Array<GuildRole> } = await response.json();
+    expect(response.status).toBe(500);
+    expect(jsonResponse).toEqual({
+      error: responseConstants.ROLE_FETCH_FAILED,
+    });
+  });
+
+  it("should return internal server error if there was any other error", async () => {
+    getGuildRoleByNameSpy.mockRejectedValueOnce({});
+
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      params: {
+        roleName: "everyone",
+      },
+      headers: { Authorization: "Bearer testtoken" },
+    });
+
+    const response: JSONResponse = await getGuildRoleByRoleNameHandler(
+      mockRequest,
+      guildEnv
+    );
+    const jsonResponse: { roles: Array<GuildRole> } = await response.json();
+    expect(response.status).toBe(500);
+    expect(jsonResponse).toEqual({
+      error: responseConstants.INTERNAL_SERVER_ERROR,
+    });
+  });
+
+  it("should return object of id and name corresponding to the role name recieved", async () => {
+    const expectedResponse = {
+      id: "role_id_one",
+      name: "everyone",
+    };
+
+    getGuildRoleByNameSpy.mockResolvedValueOnce(expectedResponse);
+
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      params: {
+        roleName: "everyone",
+      },
+      headers: { Authorization: "Bearer testtoken" },
+    });
+
+    const response: JSONResponse = await getGuildRoleByRoleNameHandler(
+      mockRequest,
+      guildEnv
+    );
+    const jsonResponse: { roles: Array<GuildRole> } = await response.json();
+    expect(response.status).toBe(200);
+    expect(jsonResponse).toEqual(expectedResponse);
+  });
+});
```

Comment:
Great work with testing a wide range of scenarios!

Reply:
Thank you